lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -236,45 +253,52 @@ def self.upload_binary(app_id, binary_path)
           request.headers["Authorization"] = "Bearer " + auth_token
         end
       rescue Faraday::ResourceNotFound
-        UI.crash!(ErrorMessage::INVALID_APP_ID)
+        UI.crash!("#{ErrorMessage::INVALID_APP_ID}: #{app_id}")
       rescue Errno::ENOENT
-        UI.crash!(ErrorMessage::APK_NOT_FOUND)
+        UI.crash!("#{ErrorMessage::APK_NOT_FOUND}: #{binary_path}")
       end
 
+      # Uploads the binary
+      #
+      # Returns the id of the release. Only happens on a successful release, on a fail it crashes.
       def self.upload(app_id, binary_path)
         upload_token = get_upload_token(app_id, binary_path)
-        status = upload_status(app_id, upload_token)
-        if status == "SUCCESS"
+        upload_status_response = get_upload_status(app_id, upload_token)
+        if upload_status_response.success?
```

Comment:
Should this be `upload_status_response.already_uploaded?` (with a corresponding check for `ALREADY_UPLOADED` in the response class?)

Reply:
(I'm not sure I know the difference between SUCCESS vs. ALREADY_UPLOADED would occur)

Reply:
a success before we start the upload process means that the upload has already occurred and completed successfully. We could add "already_uploaded?", but it would be doing the same check as "success?"